otx/api/usecases/exportable_code/prediction_to_annotation_converter.py

```diff
@@ -472,30 +498,35 @@ def convert_to_annotation(self, predictions: tuple, metadata: Dict[str, Any]) ->
             AnnotationSceneEntity: OTX annotation scene entity object.
         """
         annotations = []
-        for score, class_idx, _, mask in zip(*predictions):
-            mask = mask.astype(np.uint8)
-            contours, hierarchies = cv2.findContours(mask, cv2.RETR_CCOMP, cv2.CHAIN_APPROX_SIMPLE)
-            if hierarchies is None:
-                continue
-            for contour, hierarchy in zip(contours, hierarchies[0]):
-                if hierarchy[3] != -1:
-                    continue
-                if len(contour) <= 2 or cv2.contourArea(contour) < 1.0:
+        height, width, _ = metadata["original_shape"]
+        shape: Union[Polygon, Ellipse]
+        for score, class_idx, box, mask in zip(*predictions):
+            if self.use_ellipse_shapes:
+                shape = convert_bbox_to_ellipse(box[0] / width, box[1] / height, box[2] / width, box[3] / height)
+            else:
+                mask = mask.astype(np.uint8)
+                contours, hierarchies = cv2.findContours(mask, cv2.RETR_CCOMP, cv2.CHAIN_APPROX_SIMPLE)
+                if hierarchies is None:
                     continue
-                points = [
-                    Point(
-                        x=point[0] / metadata["original_shape"][1],
-                        y=point[1] / metadata["original_shape"][0],
-                    )
-                    for point in cv2.boxPoints(cv2.minAreaRect(contour))
-                ]
-                polygon = Polygon(points=points)
-                annotations.append(
-                    Annotation(
-                        polygon,
-                        labels=[ScoredLabel(self.labels[int(class_idx) - 1], float(score))],
-                    )
+                for contour, hierarchy in zip(contours, hierarchies[0]):
+                    if hierarchy[3] != -1:
+                        continue
+                    if len(contour) <= 2 or cv2.contourArea(contour) < 1.0:
+                        continue
+                    points = [
+                        Point(
+                            x=point[0] / width,
+                            y=point[1] / height,
+                        )
+                        for point in cv2.boxPoints(cv2.minAreaRect(contour))
+                    ]
+                    shape = Polygon(points=points)
+            annotations.append(
```

Comment:
This part is also needed to change

Reply:
Thanks ! https://github.com/openvinotoolkit/training_extensions/pull/2172/commits/c78636a7a6a99d87756ac45ddd725b3515a36621